### DISCUSSION TITLE
Error when importing MsSQLOperator on Airflow 2.5.1

### DISCUSSION THREAD

**Discussion by MariusCostin** [apache/airflow#30768] [Q&A]
### Apache Airflow Provider(s)

microsoft-mssql

### Versions of Apache Airflow Providers

apache-airflow-providers-microsoft-mssql==3.3.2

### Apache Airflow version

2.5.1

### Operating System

Linux

### Deployment

Amazon (AWS) MWAA

### Deployment details

My requirements.txt file looks like this:
pika==1.2.0
pyarrow==9.0.0
fastparquet==0.4.0
fsspec==2022.11.0
s3fs
google-cloud-bigquery==2.34.4
boto3==1.26.51
pyodbc==4.0.35
apache-airflow-providers-common-sql==1.3.3
apache-airflow-providers-microsoft-mssql==3.3.2
apache-airflow-providers-odbc==3.2.1
apache-airflow-providers-mysql==4.0.0
apache-airflow-providers-mongo==3.1.1
apache-airflow-providers-samba==4.1.0
pymssql==2.2.7
mysql-connector-python==8.0.32
mysqlclient==2.1.1
pymongo==3.13.0
dnspython==2.3.0
db-dtypes==1.0.5
duckdb
pysmb
smbprotocol==1.10.1
pymysql
openpyxl==3.0.10
simple-salesforce==1.12.3
polars

### What happened

When I upgrade the version of AWS MWAA from 2.2.2 to 2.5.1, I get the following error for all the dags that are using the MsSQLOperator:
ModuleNotFoundError: No module named 'airflow.providers.microsoft.mssql.operators'

I import the library like this: 
from airflow.providers.microsoft.mssql.operators.mssql import MsSqlOperator, MsSqlHook

The dags are working fine in the 2.2.2. environment and the requirement file is updated as the constraints show.

I also see in the logs that the provider was installed correctly.

### What you think should happen instead

I should be able to import the provider and not get an error when I migrate the dags from 2.2.2. to 2.5.1

### How to reproduce

I created the environment for Airflow MWAA 2.5.1 with the same folder for the dags as the 2.2.2. version. I used the above requirement file.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by eladkal** [comment]
> I should be able to import the provider and not get an error when I migrate the dags from 2.2.2. to 2.5.1

Not true.
Providers are independent packages from Airflow core. You should go through the provider change logs and see if there breaking changes.

You mention you do:
`from airflow.providers.microsoft.mssql.operators.mssql import MsSqlOperator, MsSqlHook`
This could have not worked in any version. You probably reported a wrong import in this issue.

The imports were not changed:
```
    from airflow.providers.microsoft.mssql.hooks.mssql import MsSqlHook
    from airflow.providers.microsoft.mssql.operators.mssql import MsSqlOperator
```
    
 If this doesn't work for you then contact MWAA support to figure why the provider is not installed as you expect.
 Note that MsSqlOperator is deprecated. You should switch to `SQLExecuteQueryOperator`
 https://github.com/apache/airflow/blob/78b8ea2f22239db3ef9976301234a66e50b47a94/airflow/providers/microsoft/mssql/operators/mssql.py#L62
